sources/osgShader/Compiler.js

```diff
@@ -953,20 +952,9 @@ define( [
                 if ( !hasShadows ) {
                     hasShadows = true;
                     this._vertexShader.push( 'uniform mat4 ModelWorldMatrix;' );
+                    this._vertexShader.push( 'varying vec4 WorldPosition;' );
+                    break;
                 }
-
-                var shadowTextureUniforms = shadowTexture.getOrCreateUniforms( i );
-                var viewMat = shadowTextureUniforms.ViewMatrix;
-                var projMat = shadowTextureUniforms.ProjectionMatrix;
-                var depthRange = shadowTextureUniforms.DepthRange;
-                var mapSize = shadowTextureUniforms.MapSize;
-                // uniforms
-                this._vertexShader.push( 'uniform mat4 ' + projMat.getName() + ';' );
-                this._vertexShader.push( 'uniform mat4 ' + viewMat.getName() + ';' );
-                this._vertexShader.push( 'uniform vec4 ' + depthRange.getName() + ';' );
-                this._vertexShader.push( 'uniform vec4 ' + mapSize.getName() + ';' );
-                // varyings
-                this._vertexShader.push( 'varying vec4 ' + shadowTexture.getVaryingName( 'VertexProjected' ) + ';' );
                 hasShadows = true;
```

Comment:
true, refactored code accordingly and more readabilidability with quick exits and less nested if/then